src/elements/fields/CheckboxField.jsx

```diff
@@ -1,6 +1,155 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import Form from 'react-bootstrap/Form';
 
+// Draws a checkmark, similar in dimensions to the default Chrome checkbox, in CSS
+const checkmarkClipPath =
+  'polygon(40% 85%, 89% 23%, 76% 12%, 39% 59%, 23% 44%, 12% 55%)';
+
+const MIN_CHECKBOX_PX = 13;
+
+const scaleCheckboxSize = (fontSize) => {
+  return Math.max(Math.floor(fontSize * 0.5), MIN_CHECKBOX_PX);
+};
+
+const applyHeightAndWidthByFontSize = (applyStyles, target) => {
+  applyStyles.apply(target, ['font_size'], (fontSize) => {
+    const scaledSize = `${scaleCheckboxSize(fontSize)}px`;
+    return { height: scaledSize, width: scaledSize };
+  });
+};
+
+const applyCheckmarkByFontSize = (applyStyles, target, colorProperty) => {
+  applyStyles.apply(target, ['font_size', colorProperty], (fontSize, color) => {
+    const scaledSize = `${scaleCheckboxSize(fontSize)}px`;
+    return {
+      boxShadow: `inset ${scaledSize} ${scaledSize} #${color}`,
+      clipPath: checkmarkClipPath
+    };
+  });
+};
+
+const applyCheckmark = (applyStyles, target, colorProperty) => {
+  applyStyles.apply(
+    target,
+    ['height', 'height_unit', 'width', 'width_unit', colorProperty],
+    (height, heightUnit, width, widthUnit, color) => {
+      return {
+        boxShadow: `inset ${width}${widthUnit} ${height}${heightUnit} #${color}`,
+        clipPath: checkmarkClipPath
+      };
+    }
+  );
+};
+
+export function applyCheckboxStyles(element, applyStyles) {
+  applyStyles.addTargets(
```

Comment:
adding targets and applying styles should ideally happen here: https://github.com/feathery-org/feathery-react/blob/master/src/elements/fields/index.jsx#L30